zipline/finance/blotter.py

```diff
@@ -77,7 +78,8 @@ def __repr__(self):
     def set_date(self, dt):
         self.current_dt = dt
 
-    def order(self, sid, amount, style, order_id=None):
+    def order(self, sid, amount, style, cancel_policy=NeverCancel(),
```

Comment:
@ehebert yes.  The long-term plan for this in my mind is that you can call `set_cancel_policy` to set the default, but you can override it on a per-order basis in `TradingAlgorithm.order`.

w/r/t @richafrank's comment here, I don't think this should be a default.  We should be explicit when we're passing a policy here, since this is only ever called from `TradingAlgorithm.order`.

In general, if we want to provide default behaviors for user-facing APIs, we should inject the default as early as possible in the call stack, and then pass values explicitly from there on out.  Adding defaults at every layer makes it much harder to change the default later, since we have to go and find all the places where we ever passed a cancel policy.  It also makes it much harder to figure out where values are actually created.